### PR TITLE
Dealing with flakes so far

### PR DESCRIPTION
**Pull request by rustyrussell**
This removes flaky, and fixes the resulting issues (so far).

Changelog-None

### CODE REVIEW

**Review by vincenzopalazzo — APPROVED** [review]
utACK https://github.com/ElementsProject/lightning/pull/5281/commits/a255e12b6f6a61281c226c0f1294aefe58f5f34e

I have also ready a lnprototest update to remove some coop-test that are disabled right now in the code base.

File: contrib/pyln-testing/pyln/testing/utils.py [hunk 1]
```diff
@@ -146,6 +146,27 @@ def get_tx_p2wsh_outnum(bitcoind, tx, amount):
     return None
 
 
+unused_port_lock = threading.Lock()
+unused_port_set = set()
+
+
+def reserve_unused_port():
```
Comment (cdecker) [inline contrib/pyln-testing/pyln/testing/utils.py:153]: One alternative thing we could do is to internalize the `ephemeral_port_reserve` lib (it's just a couple of lines), modify the reservation code not to set `REUSEADDR` itself, so it won't re-reserve a port that is in state `TIME_WAIT`, and then have `lightningd` itself leave the port in state `TIME_WAIT` on shutdown. As long as the time a node is down isn't longer than the `/proc/sys/net/ipv4/tcp_fin_timeout` we should not get the same port when calling `reserve` again.

However this method here is likely just as good, as long as we remember not to use `ephemeral_port_reserve.reserve` directly.

File: lightningd/chaintopology.c [hunk 2]
```diff
@@ -1107,7 +1109,9 @@ void setup_topology(struct chain_topology *topo,
 	start_fee_estimate(topo);
 
 	/* Once it gets initial block, it calls io_break() and we return. */
-	io_loop_with_timers(topo->ld);
+	ret = io_loop_with_timers(topo->ld);
+	assert(ret == topo);
+	log_debug(topo->ld->log, "io_loop_with_timers: %s", __func__);
```
Comment (cdecker) [inline lightningd/chaintopology.c:1114]: I assume this is ok since a failing `assert` would include `__fund__` anyway.

**Review by cdecker — APPROVED** [review]
Very nice, thanks for these fixes :+1:

ACK https://github.com/ElementsProject/lightning/pull/5281/commits/a255e12b6f6a61281c226c0f1294aefe58f5f34e

**Comment by rustyrussell** [comment]
Rebased to fix typo...

**Comment by rustyrussell** [comment]
Rebased on master again: pushing caused new flakes due to updated master.

Ack https://github.com/ElementsProject/lightning/commit/b83eae5a7de503c52e81a9cb357b86159929db2a

**Comment by rustyrussell** [comment]
Rebase again, and two more gossip flakes...

**Comment by vincenzopalazzo** [comment]
Good Job! Thanks to do that!

**Pull request merged**